src/gymnasium_interface/Pycram_gym_env.py

```diff
@@ -0,0 +1,75 @@
+import gymnasium as gym
+from gymnasium.spaces import Discrete
+from .task_executor import PyCRAMTaskExecutor
+from pycram.process_module import simulated_robot
+import sys
+
+class PyCRAMGymEnv(gym.Env):
+    def __init__(self, actions, default_params=None, objects=None, reward_function=None):
+        """
+        Initializes the Gymnasium environment.
+
+        Args:
+            actions (list): List of valid action names (e.g., ["navigate", "pick_up", "place"]).
+            default_params (dict): Default parameters for actions if not provided by the user.
+            objects (list): List of objects to add to the environment.
+            reward_function (function): User-defined function to calculate rewards.
+        """
+        super().__init__()
+        self.actions = actions
+        self.default_params = default_params or {}
+        self.objects = objects or []
+        self.reward_function = reward_function
+
+        # Dynamically define the action space
+        self.action_space = Discrete(len(actions))
+
+        # Initialize the task executor
+        self.executor = PyCRAMTaskExecutor()
+
+        # Initialize the state
+        self.state = None
+        self.reset()
+
+    def reset(self):
+        """Resets the environment."""
+        with simulated_robot:
+            self.executor.reset_task(self.objects)
+            self.state = self.executor.get_current_state()
+        return self.state, {}
+
+    def step(self, action, params=None):
```

Comment:
Please add type hints to the parameter as well as the parameter doc to the doc string.